gpiozero/pins/native.py

```diff
@@ -169,12 +292,23 @@ class NativeFactory(LocalPiFactory):
     """
     def __init__(self):
         super(NativeFactory, self).__init__()
-        self.mem = GPIOMemory()
+        queue = Queue()
+        self.mem = GPIOMemory(self.pi_info.soc)
+        self.fs = GPIOFS(self, queue)
```

Comment:
Why does it now create both `self.mem` and `self.fs` when previously it only created `self.mem` ?

Reply:
Ah - convoluted story. Previously there was GPIOMemory and GPIOFS, but I was only using GPIOMemory because I thought I could get away with just using the GPIO registers. For a rough'n'ready implementation it sort-of worked, although the edge detection was just downright awful. However, it then turned out that in recent kernels, merely attempting to access the edge detection GPIO registers locked things up, so it was time to tidy up GPIOFS properly and start using epoll on the GPIO sysfs files for edge detection instead (which is what RPi.GPIO and RPIO do).

So, GPIOFS did exist previously but it was incomplete and unused. Now it's been tidied up and is in use to handle edge detection for pins (with quite an interesting threading model - I ought to give a talk on the different pin drivers threading models at some point!).